packages/grafana-ui/src/components/GraphNG/utils.ts

```diff
@@ -44,7 +45,16 @@ function applySpanNullsThresholds(frame: DataFrame) {
 
 export function preparePlotFrame(frames: DataFrame[], dimFields: XYFieldMatchers, timeRange?: TimeRange | null) {
   // apply null insertions at interval
-  frames = frames.map((frame) => applyNullInsertThreshold(frame, null, timeRange?.to.valueOf()));
+  // and change null values to configured values
+  frames = frames.map((frame) => {
+    let f = applyNullInsertThreshold({
+      frame,
+      refFieldName: null,
+      refFieldPseudoMin: timeRange?.from.valueOf(),
+      refFieldPseudoMax: timeRange?.to.valueOf(),
+    });
+    return nullToFormattedValue(f);
```

Comment:
How did we end up back in this state?

this is pretty weird because in state-timeline, the values are already strings at this point 😬 

We really want to do this as part of the first prepare in StateTimeline.  

nullToFormattedValue for sure should not be applied to everything 😬  If the null filling is applied first, the weird parts should come for "free". The issue here is that we are applying these settings long after the numeric values are gone

Reply:
I'm admittedly struggling between getting things in a state that makes the changes to `applyNullInsertThreshold` clear while also cleanly fixing state timeline. Agree that `nullToFormattedValue` shouldn't apply to everything. 

I'll hoist up `applyNullInsertThreshold` and `nullToFormattedValue` (or the analogous) there.

Reply:
I believe I have this part addressed, moved relevant code to state timeline [here](https://github.com/grafana/grafana/pull/50054/files#diff-3d696f90972ddd1656f71d1826f0f53936bfec3cbdba397ab36521360f63ae4dR398) (and line 405 as well.